### PR TITLE
NMS-13775: Add Device Config Dao layer

### PR DESCRIPTION
**Pull request by cgorantla**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13775

### CODE REVIEW

**Review by aramosvizcarra — APPROVED** [review]
Code looks good to me

**Review by aramosvizcarra — APPROVED** [review]
Looks good to me

File: features/device-config/persistence/api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <name>OpenNMS :: Features :: Device Config :: Persistence :: API</name>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
```
Comment (RangerRick) [inline features/device-config/persistence/api/pom.xml:24]: Should this really be 1.8?

Comment (cgorantla) [reply features/device-config/persistence/api/pom.xml:24]: that's common across all pom for `maven-bundle-plugin`. Guess we should update all of them together

Comment (RangerRick) [reply features/device-config/persistence/api/pom.xml:24]: sure, not a big deal, just a thing I noticed

**Review by RangerRick — APPROVED** [review]
seems reasonable to me

**Pull request merged**